tests/app/notifications/rest/test_send_notification.py

```diff
@@ -1163,3 +1168,34 @@ def test_should_allow_international_number_on_sms_notification(client, notify_db
         headers=[('Content-Type', 'application/json'), auth_header])
 
     assert response.status_code == 201
+
+
+@pytest.mark.parametrize(
+    'template_factory, to, expected_error', [
+        (sample_template_without_sms_permission, '+447700900986', 'Cannot send text messages'),
+        (sample_template_without_email_permission, 'notify@digital.cabinet-office.gov.uk', 'Cannot send emails')
+    ])
+def test_should_not_allow_notification_if_service_permission_not_set(
```

Comment:
Actually, seems like v2 has been missed out in the permissions so will add it, and also noticed need to refactor to handle one-off msgs